sqlalchemy_filters/models.py

```diff
@@ -51,12 +56,15 @@ def _get_valid_field_names(self):
 
 
 def _is_hybrid_property(orm_descriptor):
-    return orm_descriptor.extension_type == symbol('HYBRID_PROPERTY')
+    # SQLAlchemy 2 treats extension_type as an enum, not a symbol(). Enum is at sqlalchemy.ext.hybrid.HybridExtensionType
+
+    return str(orm_descriptor.extension_type) in ("symbol('HYBRID_PROPERTY')", 'HybridExtensionType.HYBRID_PROPERTY')
```

Comment:
This is a little bit hacky but I was trying to optimize for performance. Other things I thought of:

- Use `isinstance()` to see if `orm_descriptor.extension_type` is an Enum and then use `.name` to compare. `isinstance()` can be really slow in a hot path, and this gets invoked a lot for all properties on queried models
- Try/except import since `HybridExtensionType` only exists in 2+. Kind of obtuse, I thought the same "hot path" reason to exclude this one too

Happy to change for a different opinion on the right way to check this.